src/nodes/display/PassNode.js

```diff
@@ -169,35 +169,49 @@ class PassNode extends TempNode {
 
 	}
 
-	getViewZNode() {
+	getViewZNode( name = 'depth' ) {
 
-		if ( this._viewZNode === null ) {
+		const depthTextureNode = this.getTextureNode( name );
 
-			const cameraNear = this._cameraNear;
-			const cameraFar = this._cameraFar;
+		const cameraNear = this._cameraNear;
+		const cameraFar = this._cameraFar;
 
-			this._viewZNode = perspectiveDepthToViewZ( this.getTextureNode( 'depth' ), cameraNear, cameraFar );
+		if ( name === 'depth' ) {
```

Comment:
I suggest unify the cache in a lib object like we did in `getTexture()` and `getTextureNode()`.

Reply:
Perhaps turn _linearDepth and _viewZ into cache objects? ( and maybe change _nodes into _passTextureNodes to make differences between various cache objects more explicit? )